### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by lmhs**

### CODE REVIEW

**Review by st-zelenin — APPROVED** [review]

File: app.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const http = require('http');
+const config = require('./utils/config.json');
+const express = require('express');
+const bodyParser = require('body-parser');
+const fs = require('fs');
```
Comment (st-zelenin) [inline app.js]: not used

File: app.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const handlebars = require('express-handlebars');
+const logger = require('./logger');
+const morgan = require('morgan');
+const methodOverride = require('method-override');
```
Comment (st-zelenin) [inline app.js]: what is this?

File: app.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const User = require('./models/User.js');
```
Comment (st-zelenin) [inline app.js]: not used

File: app.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+app.use(morgan(':url :date[iso]', {stream: logger.winstonLogger.stream}));
+
+app.use('/blogs', blogs);
+app.use('/login', login);
+
+app.engine('handlebars', handlebars({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+app.listen(config.port, () => console.log(`Listening to ports: ${config.port}`));
+
+
```
Comment (st-zelenin) [inline app.js]: I can't find error handling

File: models/User.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  passwordHashed: String
+});
+
+// example user:
+// username: admin
+// passwordHashed: $2a$10$pkTkA7ZdaHJo4F4MO.BTSONeEdHOrmwgvyq3WumdEPdv4KiNeAZxa
+
+UserSchema.methods.comparePasswords = function (password) {
+  return bcrypt.compareSync(password, '$2a$10$pkTkA7ZdaHJo4F4MO.BTSONeEdHOrmwgvyq3WumdEPdv4KiNeAZxa');
+  // return bcrypt.compareSync(password, this.passwordHashed);
```
Comment (st-zelenin) [inline models/User.js]: uncomment the commented

File: routes/blogs.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const jwt = require('jsonwebtoken');
+const config = require('../utils/config.json');
+
+const logger = require('../logger');
+
+mongoose.connect('mongodb://localhost:27017/frontcamp');
```
Comment (st-zelenin) [inline routes/blogs.js]: move to config

File: routes/blogs.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const jwt = require('jsonwebtoken');
+const config = require('../utils/config.json');
```
Comment (st-zelenin) [inline routes/blogs.js]: remove from utils - put next by app.js

File: routes/blogs.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    res.render('articles', {articles});
+  });
+});
+
+function checkAuth(req, res, next) {
+  if (req.get('Authorization') &&
+    req.get('Authorization').split(' ').length > 0 &&
+    req.get('Authorization').split(' ')[0] === 'Bearer' &&
+    req.get('Authorization').split(' ')[1]) {
+    const token = req.get('Authorization').split(' ')[1];
```
Comment (st-zelenin) [inline routes/blogs.js]: you repeat `req.get('Authorization').split(' ')` 4 times

File: routes/blogs.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+  .catch(err => {
+    next(err);
+  });
+});
+
+router.post('/', (req, res, next) => {
+  if (!req.body.title ||
+    !req.body.author ||
+    !req.body.content) {
+    throw new Error('empty fields');
```
Comment (st-zelenin) [inline routes/blogs.js]: error handling middle-ware will not catch this.. I would suggest `next('empty fields')`

File: routes/blogs.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+  const removeArticle = removePostById(req.params.id);
+  removeArticle.then(() => {
+    res.render('message', {text: 'Success!'});
+  })
+  .catch(err => {
+    next(err);
+  });
+});
+
+router.use(logger.errorHandler);
```
Comment (st-zelenin) [inline routes/blogs.js:124]: here you are - the error handler! usually it is in `app.js`, but it's fine to have separate handlers per route

File: routes/login.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
+const express = require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
```
Comment (st-zelenin) [inline routes/login.js]: not used

File: routes/login.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+router.post('/', authenticate, (req, res) => {
+  res.redirect('blogs/');
+});
+
+function authenticate(req, res, next) {
+  User.findOne({
+    username: req.body.username
+  }, (error, user) => {
+    if (error) {
+      throw error;
```
Comment (st-zelenin) [inline routes/login.js]: i would suggest `next(error)` - let the error-handler manage errors

**Review by st-zelenin — APPROVED** [review]

File: app.js (hunk 2 above)
Comment (lmhs) [reply app.js]: Old stuff left from previous tryouts. Will remove

File: routes/blogs.js (hunk 9 above)
Comment (lmhs) [reply routes/blogs.js]: image
Error handler caught it, but anyway changed to ```return next('empty fields')``` for consistency

**Pull request merged**